extensions/amp-accordion/0.1/amp-accordion.js

```diff
@@ -221,40 +228,156 @@ class AmpAccordion extends AMP.BaseElement {
       }
-    }, section);
+    } else { // Toggle without animation
+      this.mutateElement(() => {
+        if (toExpand) {
+          section.setAttribute('expanded', '');
+          header.setAttribute('aria-expanded', 'true');
+          // if expand-single-section is set, only allow one <section> to be
+          // expanded at a time
+          if (this.element.hasAttribute('expand-single-section')) {
+            this.sections_.forEach(sectionIter => {
+              if (sectionIter != section) {
+                sectionIter.removeAttribute('expanded');
+                sectionIter.children[0].setAttribute('aria-expanded', 'false');
+              }
+            });
+          }
+        } else {
+          section.removeAttribute('expanded');
+          header.setAttribute('aria-expanded', 'false');
+        }
+      }, section);
+    }
     this.currentState_[contentId] = !isSectionClosedAfterClick;
     this.setSessionState_();
   }
 
+  /**
+   * @param {!Element} section
+   * @return {!Promise}
+   * @private
+   */
+  animateExpand_(section) {
+    let height, duration;
+    const sectionChild = section.children[1];
+
+    return this.mutateElement(() => {
+      section.setAttribute('expanded', '');
+      st.setStyles(sectionChild, {
+        opacity: 0,
+      });
+    }).then(() => {
+      return this.measureMutateElement(
+          () => {
+            height = sectionChild./*OK*/offsetHeight;
+            const viewportHeight = this.getViewport().getSize().height;
+            duration = this.getTransitionDuration_(Math.abs(height),
+                viewportHeight);
+          },
+          () => {
+            st.setStyles(sectionChild, {
+              'opacity': '',
+              'height': 0,
+            });
+          });
+    }).then(() => {
+      return Animation.animate(this.element, tr.setStyles(sectionChild, {
+        'height': tr.px(tr.numeric(0, height)),
+      }), duration)
+          .thenAlways(() => {
+            this.mutateElement(() => {
+              st.setStyles(sectionChild, {
+                height: '',
+              });
+            });
+          });
+    });
+  }
+
+  /**
+   * @param {!Element} section
+   * @return {!Promise}
+   * @private
+   */
+  animateCollapse_(section) {
+    let height, duration;
+    const sectionChild = section.children[1];
+    return this.measureElement(() => {
+      height = section./*OK*/offsetHeight;
+      const viewportHeight = this.getViewport().getSize().height;
+      duration = this.getTransitionDuration_(Math.abs(height),
+          viewportHeight);
+    }).then(() => {
+      return Animation.animate(sectionChild, tr.setStyles(sectionChild, {
+        'height': tr.px(tr.numeric(height, 0)),
+      }), duration).thenAlways(() => {
+        return this.mutateElement(() => {
+          section.removeAttribute('expanded');
+        });
+      }).then(() => {
+        return this.mutateElement(() => {
+          st.setStyles(sectionChild, {
```

Comment:
Can this be part of the `mutateElement` callback in the `thenAlways` block? I can't tell what would be affected by setting the height a mutation cycle after removing the attribute. If it is needed, a comment explaining why could be helpful for future editors.

Reply:
This was a bit stupid of me. I think this was leftover from an incomplete revert of the triggerEvent feature.

Reply:
not stupid! refactors are hard : ) that's why we code review